### PR TITLE
Deepp Volume adapter creation.

### PR DESCRIPTION
**Pull request by MercuryPunk**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
DeeppDAO

##### Twitter Link:
https://twitter.com/deeppxyz

##### Website Link:
https://deepp.xyz/

##### Logo (High resolution, will be shown with rounded borders):
https://deepp.xyz/images/DeeppDAO_SYMBOL.png

##### Chain:
Arbitrum

##### Short Description (to be shown on DefiLlama):
Prediction market. Non custodial. Transparent. Trustless. Community owned.

##### Token address and ticker if any:
0x9773367fB87CFbcB7F6A0AB9a9e3Ab635023b7Da

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Prediction Market

##### methodology (what is being counted as tvl, how is tvl being calculated):
Lists the number of owned USDC tokens in the Deepp LP and BetLock contracts.

##### Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/deepp exports TVL: 
        
 
 
    arbitrum                  329.86 k
    
    total                    329.86 k

**Comment by g1nt0ki** [comment]
hi @MercuryPunk , why is the `fullTokenBalance` value higher than token amount in the contract?

**Comment by MercuryPunk** [comment]
Because it list the balance in the the Actual LP + the amount of tokens temporarily locked in bets in the betlock contract on behalf of LP contract. You have changed the logic, so now a different amount is displayed instead.
When bets are settled, some are won by the betters and will decrease the combined balance of the LP contract. Others are lost by the betters and will be returned to the LP contracts direct balance. When that happens the LP contract will receive both its own stake and the betters stake for the bet. 
Until a bet is settled, the best way to estimate the LPs full balance is to add its own direct balance and its own combined stakes for all bets that have not been settled yet. The betlock contract holds the balance of both sides of the bet, until a bet settles.

So in other words you refactor has changed the logic so now the wrong number will be displayed. 
I can give you the betlock contract address as well, but adding the combined USDC token balance of the 2 contracts will also result in the wrong number. That would assume, that all betters will lose their unsettled bets, in order for the LP receive the fully locked amount after settling.

I am not sure how you want to proceed from here?

**Comment by g1nt0ki** [comment]
hi @MercuryPunk I merged my change, so it gets listed while we still discuss the tvl. Yeah, can you share the betlock contract, I will include value there as well, tvl is all value locked in product protocols, we dont differentiate between LP & betters

**Comment by MercuryPunk** [comment]
OK, fair enough - they are all tokens temporarily owned be the protocol, at least until bets settles.
Here is the Betlock contract address:
BET_BOX_ADDRESS=0x05E1F51067a3Af2f9d0994a97779a78a2E26f921

If you want to add the full protocol balance of USDC even temporarily owned tokens,
we also have 4 separate contracts that accumulates the fees paid from bets and from LP 
token buy/sell transactions. They are stored in the 4 contracts below, until they are
distributed/claimed as rewards by the protocol users (only 3 of them holds fees so far).
Can you add those as well, if you don't mind?

BET_FEE_HANDLER_1_ADDRESS=0x42e27a7D424C22ED3658970CAB01260f8C0EC5Bc
BET_FEE_HANDLER_2_ADDRESS=0xD5750d44D9F2ed117FB3441D80B423acD7634Cf5
LP_FEE_HANDLER_1_ADDRESS=0x80e674e6277A7e9073A6Dff5C98e08816c0D73f5
LP_FEE_HANDLER_2_ADDRESS=0xAb5e18D29C20709954eA380e998FFFeaeB4FF691

Thanks for your help.
Br.

**Comment by g1nt0ki** [comment]
updated here: https://github.com/DefiLlama/DefiLlama-Adapters/commit/a745a4ccb345145f9dde37c4efd31c381cd156f1
